CMakeLists.txt

```diff
@@ -1,6 +1,30 @@
+cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+
+if(POLICY CMP0025)
+  cmake_policy(SET CMP0025 NEW)
+endif()
+
 project(MONGO_CXX_DRIVER LANGUAGES CXX)
 
-cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
+    message(FATAL_ERROR "Insufficient GCC version - GCC 4.9+ required")
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "19.0.23506")
+    message(FATAL_ERROR "Insufficient Microsoft Visual C++ version - MSVC 2015U1+ required")
```

Comment:
Is it possible to include the current version in this message?  And would that be helpful to users?